scripts/exportlegends.lua

```diff
@@ -236,6 +236,72 @@ function export_more_legends_xml()
     end
     file:write("</entities>\n")
 
+    file:write("<poetic_forms>\n")
+    for formK, formV in ipairs(df.global.world.poetic_forms.all) do
+        file:write("\t<poetic_form>\n")
+        file:write("\t\t<id>"..formV.id.."</id>\n")
+        file:write("\t\t<name>"..dfhack.df2utf(dfhack.TranslateName(formV.name,1)).."</name>\n")
+        file:write("\t</poetic_form>\n")
+    end
+    file:write("</poetic_forms>\n")
+
+    file:write("<musical_forms>\n")
+    for formK, formV in ipairs(df.global.world.musical_forms.all) do
+        file:write("\t<musical_form>\n")
+        file:write("\t\t<id>"..formV.id.."</id>\n")
+        file:write("\t\t<name>"..dfhack.df2utf(dfhack.TranslateName(formV.name,1)).."</name>\n")
+        file:write("\t</musical_form>\n")
+    end
+    file:write("</musical_forms>\n")
+
+    file:write("<dance_forms>\n")
+    for formK, formV in ipairs(df.global.world.dance_forms.all) do
+        file:write("\t<dance_form>\n")
+        file:write("\t\t<id>"..formV.id.."</id>\n")
+        file:write("\t\t<name>"..dfhack.df2utf(dfhack.TranslateName(formV.name,1)).."</name>\n")
+        file:write("\t</dance_form>\n")
+    end
+    file:write("</dance_forms>\n")
+
+    file:write("<written_contents>\n")
+    for wcK, wcV in ipairs(df.global.world.written_contents.all) do
+        file:write("\t<written_content>\n")
+        file:write("\t\t<id>"..wcV.id.."</id>\n")
+        file:write("\t\t<title>"..wcV.title.."</title>\n")
+        file:write("\t\t<page_start>"..wcV.page_start.."</page_start>\n")
+        file:write("\t\t<page_end>"..wcV.page_end.."</page_end>\n")
+        for refK, refV in pairs(wcV.refs) do
+            file:write("\t\t<reference>\n")
+            file:write("\t\t\t<type>"..df.general_ref_type[refV:getType()].."</type>\n")
+            if refV:getType() == 64     then file:write("\t\t\t<id>"..refV.anon_1.."</id>\n") -- written content
```

Comment:
This section should use checks like `if refV:getType() == df.general_ref_type.WRITTEN_CONTENT` instead. You can get the appropriate names in the `lua` interpreter with `~df.general_ref_type[64]`, etc. (or use `@df.general_ref_type` for a complete list).